### PR TITLE
Init state logic

### PR DESCRIPTION
**Pull request by RobertGemmaJr**
**colorMap**
- The `colorMap` prop if given
- The first property in `colorMaps` if `colorMap` is not given but `colorMaps` is
- `DEFAULT_COLOR_MAP` (grayscale) if neither `colorMap` nor `colorMaps` are given.

**colorMaps**
I added some logic to make sure `colorMap` is added to `colorMaps` if the user doesn't do so. And with that knowledge:

- The `colorMaps` prop + `DEFAULT_COLOR_MAPS` if `colorMaps` is given and `useDefaultColorMaps` is true
- Just `colorMaps` if `colorMaps` is given and `useDefaultColorMaps` is false. 
- Just `DEFAULT_COLOR_MAPS` if `colorMaps` is not given and `useDefaultColorMaps` is true.
- Just `DEFAULT_COLOR_MAP` (`{default: grayscale}`) if `colorMaps` is not given and `useDefaultColorMaps` is false.

### CODE REVIEW

**Comment by RobertGemmaJr** [comment]
Wait for PR #43 to be merged before merging this branch

**Review by broarr — APPROVED** [review]
This looks good to me. I agree with @mcmcgrath13 on the used internally stuff. I wonder if there's a way we could no export that stuff as props?

File: README.md [hunk 1]
```diff
@@ -24,26 +24,42 @@ CSS styling for the height must be provided and a custom width can be provided a
...
-  colorMap: PropTypes.string,
   /**
-   * Dictionary of color maps available in the controls.
-   *  key: Name of the color map
-   *  value: Path to the color map
+   * The current color map applied by the transferFunction
+   * It will default to the first object in colorMaps if no colorMap is provided
+   * It will default to grayscale if neither colorMap nor colorMaps is provided.
+   *
+   *  name: Common name of the color map - used internally
```
[the remark this answers is not in the document]
Comment (RobertGemmaJr) [reply README.md:32]: Yeah, `VolumeViewer.jsx` will add the given `colorMap` to `colorMaps` if it's not present.

**Comment by RobertGemmaJr** [comment]
Okay so as far as the `used internally` stuff. That was referring to the `colorMap.name` property because it was used as the value of the Reach-UI `ListboxInput` dropdown. I suppose the comment is a little confusing so I just deleted it.

That said, I could make `colorMaps` just the path instead of an object with .name and .path. Is that a better design? Or should I just leave it as an object? Changing it to just a string would be a breaking change

**Comment by mcmcgrath13** [comment]
> Okay so as far as the `used internally` stuff. That was referring to the `colorMap.name` property because it was used as the value of the Reach-UI `ListboxInput` dropdown. I suppose the comment is a little confusing so I just deleted it.
> 
> That said, I could make `colorMaps` just the path instead of an object with .name and .path. Is that a better design? Or should I just leave it as an object? Changing it to just a string would be a breaking change

Isn't the name also used in the display of the color map? and as the alt text?

**Pull request merged**